py-polars/polars/cfg.py

```diff
@@ -14,7 +14,7 @@ def set_utf8_tables(cls) -> "Type[Config]":
         """
         Use utf8 characters to print tables
         """
-        os.environ.unsetenv("POLARS_FMT_NO_UTF8")  # type: ignore
+        os.environ["POLARS_FMT_NO_UTF8"] = ""
```

Comment:
I don't think this is the same. We check if the `env var` exists from the rust side. Now it does exist, where we first removed it.

Reply:
I should have added a comment here when making this change, as it is not obvious. 

There are a couple of things going on here:
1. `unsetenv` lives in the documentation on `os`, although it seems it also lives on `os.environ`. I think this is also why `mypy` otherwise complains, it doesn't validate.
2. on Windows, it is only implemented from Python 3.9 onwards, see https://docs.python.org/3.10/library/os.html#os.unsetenv
3. the recommendation in the docs say it is probably best to do `del os.environ["POLARS_FMT_NO_UTF8"]`
4. in the bug tracker, it is pointed out that it is implemented in Python 3.8 as I did here, see top comment on https://bugs.python.org/issue39413

Overall, I would say the better fix is to not rely on removing an ENV variable, but explicitly set 0 or 1. I.e., below there is another config that says this:

```python
 @classmethod
    def set_utf8_tables(cls) -> "Type[Config]":
        """
        Use utf8 characters to print tables
        """
        os.environ.unsetenv("POLARS_FMT_NO_UTF8")  # type: ignore
        return cls

    @classmethod
    def set_ascii_tables(cls) -> "Type[Config]":
        """
        Use ascii characters to print tables
        """
        os.environ["POLARS_FMT_NO_UTF8"] = "1"
        return cls
```

it would be more consistent if we could do `os.environ["POLARS_FMT_NO_UTF8"] = "0" in what I changed above, but Im not sure how things are handled on the Rust side.

Reply:
At the moment it just checks if this environment variable exists.
```rust
./polars/polars-core/src/fmt.rs
369-        #[cfg(feature = "pretty_fmt")]
370-        {
371-            let mut table = Table::new();
372:            let preset = if std::env::var("POLARS_FMT_NO_UTF8").is_ok() {
373-                ASCII_FULL
374-            } else {
375-                UTF8_FULL
```

Reply:
I have reverted this, the initial goal of removing the `type: ignore` becomes muddied with the inconsistent handling between the various configs; I feel that is better tackled in a separate PR.